libdevcore/JsonUtils.cpp

```diff
@@ -31,3 +31,86 @@ void dev::validateFieldNames(json_spirit::mObject const& _obj, std::set<std::str
 			BOOST_THROW_EXCEPTION(UnknownField() << errinfo_comment(comment));
 		}
 }
+
+std::string dev::jsonTypeAsString(json_spirit::Value_type _type)
+{
+    switch (_type)
+    {
+    case json_spirit::obj_type:
+        return "json Object";
+    case json_spirit::array_type:
+        return "json Array";
+    case json_spirit::str_type:
+        return "json String";
+    case json_spirit::bool_type:
+        return "json Bool";
+    case json_spirit::int_type:
+        return "json Int";
+    case json_spirit::real_type:
+        return "json Real";
+    case json_spirit::null_type:
+        return "json Null";
+    default:
+        return "json n/a";
+    }
+}
+
+void dev::requireJsonFields(json_spirit::mObject const& _o, std::string const& _config,
+    std::map<std::string, jsonType> const& _validationMap)
+{
+    // check for unexpected fiedls
+    for (auto const& field : _o)
+    {
+        if (!_validationMap.count(field.first))
+        {
+            std::string const comment =
+                "Unexpected field '" + field.first + "' in config: " + _config;
+            std::cerr << comment << "\n"
+                      << json_spirit::write_string((json_spirit::mValue)_o, true) << "\n";
+            BOOST_THROW_EXCEPTION(UnknownField() << errinfo_comment(comment));
+        }
+    }
+
+    // check field types with validation map
+    for (auto const vmap : _validationMap)
+    {
+        // check that all required fields are in the object
+        if (!_o.count(vmap.first))
+        {
+            if (vmap.second.second == jsonField::Required)
+            {
+                std::string const comment =
+                    "Expected field '" + vmap.first + "' not found in config: " + _config;
+                std::cerr << comment << "\n"
+                          << json_spirit::write_string((json_spirit::mValue)_o, true) << "\n";
+                BOOST_THROW_EXCEPTION(MissingField() << errinfo_comment(comment));
+            }
+            else if (vmap.second.second == jsonField::Optional)
+                continue;
+        }
+
+        // check that field type is one of allowed field types
+        bool matched = false;
+        for (auto const& type : vmap.second.first)
+        {
+            if (_o.at(vmap.first).type() == type)
+                matched = true;
+        }
+        if (matched == false)
+        {
+            std::string sTypes;
+            for (auto const& type : vmap.second.first)
```

Comment:
I'd use `boost::algorithm::join` here instead of reinventing joining the strings, but it's up to you